src/main/scala/elevate/rise/rules/algorithmic.scala

```diff
@@ -28,42 +30,79 @@ object algorithmic {
       case App(Map(), f) => Success((split(n) >> map(map(f)) >> join) :: e.t)
       case _             => Failure(splitJoin(n))
     }
-    override def toString = s"splitJoin($n)"
+    override def toString: String = s"splitJoin($n)"
   }
 
   // fusion / fission
 
   def mapFusion: Strategy[Rise] = `*g >> *f -> *(g >> f)`
   case object `*g >> *f -> *(g >> f)` extends Strategy[Rise] {
     def apply(e: Rise): RewriteResult[Rise] = e match {
-      case App(App(Map(), f), App(App(Map(), g), arg)) => Success(map(typed(g) >> f)(arg) :: e.t)
-      case _                                           => Failure(mapFusion)
+      case App(App(Map(), f), App(App(Map(), g), arg)) =>
+        Success(map(typed(g) >> f)(arg) :: e.t)
+      case _ => Failure(mapFusion)
     }
-    override def toString = s"mapFusion"
+    override def toString: String = "mapFusion"
   }
 
+  case object fuseReduceMap extends Strategy[Rise] {
```

Comment:
the naming is not very consistent with `mapFusion`

Reply:
also I think a quick specification like `*g >> *f -> *(g >> f)` would be useful?

Reply:
renamed the function but I think the "specifications" are useless and we should remove them.